src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/GeneratedDocumentPublisher.cs

```diff
@@ -1,16 +1,273 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.TextDifferencing;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.Logging;
 using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
 using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer;
 
-internal abstract class GeneratedDocumentPublisher : IProjectSnapshotChangeTrigger
+internal class GeneratedDocumentPublisher : IGeneratedDocumentPublisher, IProjectSnapshotChangeTrigger
```

Comment:
It looks like this can be `sealed`, based on the changes in this commit, but I'm guessing its used in mocks somewhere that are unchanged, or you would have done so, right?

I say this because there are quite a number of classes in this PR that I would have thought could be sealed, but also a non-zero number of classes that weren't sealed but you made them sealed, so maybe this is just one you missed :)

Reply:
Good point. I hadn't done a pass through my changes to look for types that can be sealed now. Will do so!